sql/logictest: TestLogic/fakedist-opt/cascade/UpdateCascade_InterleavedRestrict timed out under stress

SHA: https://github.com/cockroachdb/cockroach/commits/5a17b3dcee1f6d7db9b87e705798bac80594b27c

Parameters:
```
TAGS=
GOFLAGS=-race -parallel=1
```

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stress instead of stressrace and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
stdbuf -oL -eL \
make stressrace TESTS=TestLogic/fakedist-opt/cascade/UpdateCascade_InterleavedRestrict PKG=github.com/cockroachdb/cockroach/pkg/sql/logictest TESTTIMEOUT=5m STRESSFLAGS='-maxtime 20m -timeout 10m' 2>&1 | tee /tmp/stress.log
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=1322208&tab=buildLog

```
Slow failing tests:
TestLogic/fakedist-opt/cascade/UpdateCascade_InterleavedRestrict - 0.00s

Slow passing tests:
<none>

```

May not be exactly the same, but got something like this in CI:

```
=== CONT  TestLogic/fakedist/postgres_jsonb
--- progress: testdata/logic_test/postgresjoin: 25 statements/queries
--- progress: testdata/logic_test/privilege_builtins: 56 statements/queries
--- progress: testdata/logic_test/cascade: 208 statements/queries
--- progress: testdata/logic_test/postgresjoin: 34 statements/queries
--- progress: testdata/logic_test/sequences: 146 statements/queries
--- progress: testdata/logic_test/alter_table: 90 statements/queries
--- progress: testdata/logic_test/fk: 437 statements/queries
--- progress: testdata/logic_test/select_table_alias: 8 statements/queries
--- progress: testdata/logic_test/privilege_builtins: 62 statements/queries
--- progress: testdata/logic_test/postgresjoin: 39 statements/queries
panic: test timed out after 12m0s

goroutine 1313952 [running]:
testing.(*M).startAlarm.func1()
	/usr/local/go/src/testing/testing.go:1296 +0xfd
created by time.goFunc
	/usr/local/go/src/time/sleep.go:172 +0x44

goroutine 1 [chan receive, 12 minutes]:
testing.(*T).Run(0xc00015f300, 0x2bc875f, 0x9, 0x2d02fc0, 0x922a86)
	/usr/local/go/src/testing/testing.go:879 +0x383
testing.runTests.func1(0xc00015f200)
	/usr/local/go/src/testing/testing.go:1119 +0x78
testing.tRunner(0xc00015f200, 0xc000629de8)
	/usr/local/go/src/testing/testing.go:827 +0xbf
testing.runTests(0xc0002631a0, 0x4a09ea0, 0x5, 0x5, 0xe20cddf)
	/usr/local/go/src/testing/testing.go:1117 +0x2aa
testing.(*M).Run(0xc0004c5f00, 0x0)
	/usr/local/go/src/testing/testing.go:1034 +0x165
github.com/cockroachdb/cockroach/pkg/sql/logictest.TestMain(0xc0004c5f00)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/logictest/main_test.go:37 +0xb5
main.main()
	_testmain.go:48 +0x13d

```

https://teamcity.cockroachdb.com/viewLog.html?buildId=1321515&buildTypeId=Cockroach_UnitTests&tab=buildResultsDiv

--- a/pkg/sql/logictest/logic.go
+++ b/pkg/sql/logictest/logic.go
@@ -2083,6 +2083,10 @@ func RunLogicTest(t *testing.T, globs ...string) {
 	// Note: there is special code in teamcity-trigger/main.go to run this package
 	// with less concurrency in the nightly stress runs. If you see problems
 	// please make adjustments there.
+	// As of 6/4/2019, the logic tests never complete under race.
+	if testutils.NightlyStress() && util.RaceEnabled {
+		t.Skip("logic tests and race detector don't mix: #37993")
+	}
 
 	if skipLogicTests {
 		t.Skip("COCKROACH_LOGIC_TESTS_SKIP")

